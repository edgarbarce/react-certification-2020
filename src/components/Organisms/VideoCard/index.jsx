import React from 'react';
import styled from 'styled-components';
import Div from '../../atoms/Div';
import VideoImg from '../../atoms/VideoImg';
import HeaderMedium from '../../atoms/HeaderMedium';
import Paragraph from '../../atoms/Paragraph';

const Card = styled.div`
  display: inline-block;
  min-height: 350px;
  max-height: 350px;
  background-color: white;
  margin: 1%;
`;

function VideoCard({ imgCover, title, description }) {
  return (
    <Card>
      <Div maxWidth="320px">
        <VideoImg imgCover={imgCover} />
      </Div>
      <Div maxWidth="320px">
        <HeaderMedium fixedHeight="20px">{title}</HeaderMedium>
        <Paragraph fixedHeight="20px">{description}</Paragraph>
      </Div>
    </Card>
  );
}

export default VideoCard;
