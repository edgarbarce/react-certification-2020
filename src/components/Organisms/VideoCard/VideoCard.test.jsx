import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import VideoCard from './index';

describe('Test VideoCard Component', () => {
  afterEach(cleanup);
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<VideoCard />, div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<VideoCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
