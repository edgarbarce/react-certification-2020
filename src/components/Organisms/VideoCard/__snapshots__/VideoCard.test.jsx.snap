// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VideoCard Component matches snapshot 1`] = `
<div
  className="sc-eCssSg kUPbgp"
>
  <div
    className="sc-bdfBwQ cHyQMg"
    data-testid="divTest"
  >
    <img
      alt=""
      className="sc-gsTCUz deAlxh"
    />
  </div>
  <div
    className="sc-bdfBwQ cHyQMg"
    data-testid="divTest"
  >
    <h4
      className="sc-dlfnbm iGzhYW"
    />
    <p
      className="sc-hKgILt fEhFOs"
    />
  </div>
</div>
`;
